### ISSUE TITLE
Goto line picker should not announce the line content

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#180144]
From https://github.com/microsoft/vscode/pull/179469

**Comment by meganrogge** [comment]
> should we announce just the line and not the contents

if you mean the line number, that would not be informative as the user just typed in the line number they want to go to. the screen reader user wants to know the line content

**Comment by meganrogge** [comment]
> what about a line that is minified and 10MB in size, do we want to announce it

this is a good edge case to fix

**Comment by bpasero** [comment]
> the screen reader user wants to know the line content

Then how does it work when I use arrow keys to change the line number in the editor? Are we announcing the line to the user by alerting the contents of the line?

**Comment by meganrogge** [comment]
when I move my cursor manually, it uses the selection with reason of `render`
![Screenshot 2023-04-17 at 11 15 36 AM](https://user-images.githubusercontent.com/29464607/232574446-9bb5bfd3-d6f5-40ed-afb3-c64dd80477e6.png)
![Screenshot 2023-04-17 at 11 14 37 AM](https://user-images.githubusercontent.com/29464607/232574247-60e1e2e4-f040-4730-88a7-8f2d9260eccf.png)

**Comment by meganrogge** [comment]
we are using the selection there, not sure why it doesn't work when the position changes due to a command

https://github.com/microsoft/vscode/blob/f63c43c8c09439fde94554018b2fbb1a42ed4d80/src/vs/editor/browser/controller/textAreaState.ts#L240-L288

**Comment by meganrogge** [comment]
based on my logging, this should be working. So I think it has to be a screen reader issue

**Comment by bpasero** [comment]
For me on macOS even without selection, the entire line gets read. But when you look at https://github.com/microsoft/vscode/blob/f63c43c8c09439fde94554018b2fbb1a42ed4d80/src/vs/editor/browser/controller/textAreaState.ts#L240-L288, you can see how complicate the method is for figuring out what to `alert`. It seems to involve a lot more than just getting the line content.

**Comment by meganrogge** [comment]
> For me on macOS even without selection, the entire line gets read.

when you navigate with arrow keys? or when you use `Go to line`?

**Comment by meganrogge** [comment]
> Unrelated, when you look at https://github.com/microsoft/vscode/blob/f63c43c8c09439fde94554018b2fbb1a42ed4d80/src/vs/editor/browser/controller/textAreaState.ts#L240-L288, you can see how complicate the method is for figuring out what to alert. It seems to involve a lot more than just getting the line content.

Yes, it would be ideal to make use of that and not have to separately manage it as you've pointed out. I hope that's possible. Another reason I hope that works is `alert` does not currently do anything on macOS due to an issue with VoiceOver. Most screen reader users are on Windows, but that's not ideal.

**Comment by bpasero** [comment]
To clarify: this was using arrow keys. So yes, there seems to be a bug with goto line, if it does not work the same. Maybe an issue that has to do with how focus is in the picker and then moves to the editor.

**Comment by meganrogge** [comment]
Hopefully Alex Dima or Henning have an idea

**Comment by bpasero** [comment]
Moving to May for consideration. I think this should not lay around untackled.

**Comment by bpasero** [comment]
One difference that I can observe is that when I use arrow keys for navigation, the entire line gets this black box:

![image](https://user-images.githubusercontent.com/900690/235605967-6722a346-c9ce-4b8e-9495-eaf8ac6e59cb.png)

But when the selection changes from goto line, that box is missing:

![image](https://user-images.githubusercontent.com/900690/235606142-bdbe38ae-44d7-4a70-951f-4f0ad74e8694.png)

Similar, when I use the mouse to navigate, I also do not see the box. I am quite sure the issue is related and I even think there maybe other instances where the line is not correctly announced, not just in quick access, so I think a fix needs to go into the editor and then maybe when I call `editor.setSelection`, this box should appear?

**Comment by bpasero** [comment]
Btw you can easily reproduce this issue differently: just bind a keybinding to the `workbench.action.focusActiveEditorGroup` command, select a line, focus the explorer and invoke the command. The screen reader will not announce the line, but the entire file until the line.

**Comment by bpasero** [comment]
Btw I tested Xcode and it behaves the same how we used to: when I goto line (`Cmd+L`), the entire file is being read and not the selected line.

**Comment by bpasero** [comment]
Are we actually sure our change improved something or are we guessing? 

My main problem with this approach is that it forces all components to do the editor's accessibility. I might be building an extension that does "Go to line" and it would not be accessible because it misses the `alert` change.

I would suggest to revert the `alert` change in the picker and fix it on the level of the editor if possible or keep it as is.

**Comment by meganrogge** [comment]
We are not guessing -- `alert` fixes this issue on windows. We should revert that change when we find a different way to do this.
